scripts/HRRR/download_hrrr.sh

```diff
@@ -86,100 +95,109 @@ check_file_existence(){
   fi
   return 1
 }
+export -f check_file_existence
+
+download_hrrr() {
+  DAY_HOUR=$1
+  FC_HOUR=$2
+  FILE_NAME="hrrr.t$(printf "%02d" $DAY_HOUR)z.wrfsfcf0${FC_HOUR}.grib2"
+
+  printf "  File: ${FILE_NAME}"
+
+  check_file_existence
+  if [[ $? -eq 0 ]]; then
+    return
+  elif [[ -e ${FILE_NAME} ]]; then
+    rm ${FILE_NAME}
+  fi
+
+  check_file_in_archive ${ARCHIVE}
+
+  if [ $? -eq 1 ]; then
+    check_alternate_archive
+  fi
+
+  if [ $? -eq 1 ]; then
+    >&2 printf "  ** File not available **\n"
+
+    # Try a previous hour of the day when getting the F06 forecast
+    if [[ ${FC_HOUR} -eq 6 ]]; then
+      NEW_DATE=$(date -u -d "${DATE} $(printf "%02d" $DAY_HOUR):00:00 1 hour ago" "+%Y%m%d%H")
+      ALT_DATE=${NEW_DATE:0:-2}
+      FILE_NAME="hrrr.t${NEW_DATE:(-2)}z.wrfsfcf0$(($FC_HOUR + 1)).grib2"
+
+      >&2 printf "  ** Checking previous hour: hrrr.${ALT_DATE}/${FILE_NAME}"
+
+      check_file_existence
+      if [[ $? -eq 0 ]]; then
+        unset ALT_DATE
+        return
+      fi
+      check_file_in_archive $ARCHIVE
+
+      if [[ $? -eq 1 ]]; then
+        check_alternate_archive
+
+        if [[ $? -eq 1 ]]; then
+          >&2 printf "  ** File not present in previous hour\n"
+          return
+        fi
+      fi
+    else
+      return
+    fi
+  fi
+
+  TMP_FILE="${FILE_NAME}_tmp"
+  [[ -e ${TMP_FILE} ]] && rm $TMP_FILE
```

Comment:
Also remove any previously used `tmp` files.

Reply:
Looks good. My inclination is to add these last two lines in between lines 108 and 109 to remove any previously created TMP files as well before moving on.